src/wobjectdefs.h

```diff
@@ -711,41 +898,80 @@ constexpr auto simple_hash(char const *p) {
 #define W_INTEGRAL_CONSTANT_HELPER(NAME, ...) std::integral_constant<unsigned long long, w_internal::simple_hash(#NAME #__VA_ARGS__)>
 #endif
 
+namespace w_internal {
+
+template<size_t L, size_t N = 0, size_t M = 1024, size_t X = (N+M)/2, class F, class... As>
+constexpr auto count(F f, As... as) {
+    using R = decltype(std::declval<F>()(index<X>, As{}...));
+    if constexpr (N==X) {
+        (void)f;
+        ((void)as, ...);
+        return std::is_same_v<void, R> ? N : M;
+    }
+    else if constexpr (std::is_same_v<void, R>) {
+        return count<L, N, X>(f, as...);
+    }
+    else {
+        return count<L, X, M>(f, as...);
+    }
+}
+
+template<class T>
+constexpr void w_SlotState(IndexBase, T**);
+template<size_t L, class ThisType>
+constexpr auto w_SlotStateCount = count<L>([](auto... a) -> decltype(w_SlotState(a...)) {}, ThisType{});
```

Comment:
Since this is repeated, maybe it can be in a macro.

Or, perhaps it could be a 

```
template<class T, class S> constexpr void w_State(IndexBase, T**, S);
```

and have tags like `struct SlotState{};`

Reply:
good idea… I will try it out.